python/tvm/auto_scheduler/relay_integration.py

```diff
@@ -126,6 +126,9 @@ def extract_tasks(
                 target=target,
                 target_host=target_host,
                 hardware_params=hardware_params,
+                # When auto scheduler is used in end to end network, try to apply layout rewrite
+                # to improve the overall performance
+                layout_rewrite_option=LayoutRewriteOption.REWRITE_FOR_PRE_TRANSFORMED
```

Comment:
We can set layout rewrite options according  to the target.
For cpu/mali, we can use `REWRITE_FOR_PRE_TRANSFORMED`. For cuda, we can use `NO_REWRITE`.

We can create a common utility function to reuse these lines.
https://github.com/apache/tvm/blob/0b2ab56705a1c26aef402d1c0d48a887fc76dbd3/python/tvm/auto_scheduler/relay_integration.py#L262-L268

Reply:
Done.